### PR TITLE
Date time pickers

### PR DESCRIPTION
**Pull request by meluru**

### CODE REVIEW

File: indico/modules/rb_new/client/js/components/filters/TimeForm.jsx [hunk 1]
```diff
@@ -64,28 +62,11 @@ export default class TimeForm extends FilterFormComponent {
...
-                -
-                <TimePicker value={endTime}
-                            styleName="time-picker"
-                            {...props}
-                            onChange={(value) => {
-                                this.setTimes(startTime, value);
-                            }} />
+                <TimeRangePicker startTime={startTime}
+                                 endTime={endTime}
+                                 onChange={(start, end) => this.setTimes(start, end)} />
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/components/filters/TimeForm.jsx]: `onChange={this.setTimes}`

File: indico/modules/rb_new/client/js/components/TimeRangePicker.jsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+    generateEndTimeOptions = (start) => {
+        const options = [];
+        const end = moment().endOf('day');
+        const next = moment(start).add(30, 'm');
+        let serializedNext, duration, text;
+        while (next < end) {
+            duration = _humanizeDuration(moment.duration(next.diff(start)));
+            serializedNext = serializeTime(moment(next));
+            text = <div>{serializedNext} <div styleName="duration">({duration})</div></div>;
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/components/TimeRangePicker.jsx]: `const text = ...`; no need to define it outside, `const` and `let` are block-scoped.

File: indico/modules/rb_new/client/js/components/TimeRangePicker.module.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+.time-dropdown {
+    min-width: 90px !important;
+    margin-right: 14px;
+    text-align: center !important;
+}
+
+.duration {
+    font-size: x-small;
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/components/TimeRangePicker.module.scss]: do we really want to use that instead of a value in `em`?

Comment (meluru) [reply indico/modules/rb_new/client/js/components/TimeRangePicker.module.scss]: yeah, em seems better

**Pull request merged**